### PR TITLE
LPS-84736 Client tests for input control

### PR DESCRIPTION
**Pull request by ruben-pulido**

### CODE REVIEW

File: modules/apps/headless-apio/structured-content/structured-content-apio-impl/src/main/java/com/liferay/structured/content/apio/internal/architect/resource/StructuredContentNestedCollectionResource.java [hunk 1]
```diff
@@ -839,8 +844,25 @@ public StructuredContentField(
...
 		public String getDDMFormFieldDataType() {
 			try {
-				return _ddmStructure.getFieldDataType(
+				String dataType = _ddmStructure.getFieldDataType(
 					_ddmFormFieldValue.getName());
+
+				String displayDataType = dataType;
+
+				if (Objects.equals(dataType, "document-library")) {
+					displayDataType = "document";
```
Comment (nhpatt) [inline modules/apps/headless-apio/structured-content/structured-content-apio-impl/src/main/java/com/liferay/structured/content/apio/internal/architect/resource/StructuredContentNestedCollectionResource.java]: I don't mind having multiple returns in this case, that way you don't have to do the other variable (that you didn't have to do either way)

File: modules/apps/headless-apio/structured-content/structured-content-apio-impl/src/main/java/com/liferay/structured/content/apio/internal/architect/resource/StructuredContentNestedCollectionResource.java [hunk 2]
```diff
@@ -839,8 +844,25 @@ public StructuredContentField(
...
+				}
+				else if (Objects.equals(dataType, "journal-article")) {
```
Comment (nhpatt) [inline modules/apps/headless-apio/structured-content/structured-content-apio-impl/src/main/java/com/liferay/structured/content/apio/internal/architect/resource/StructuredContentNestedCollectionResource.java]: Any reason for using Objects.equals in particular?

File: modules/apps/headless-apio/structured-content/structured-content-apio-impl/src/main/java/com/liferay/structured/content/apio/internal/architect/resource/StructuredContentNestedCollectionResource.java [hunk 3]
```diff
@@ -854,6 +876,33 @@ public String getDDMFormFieldDataType() {
...
+
+				if (Objects.equals(type, DDMFormFieldType.CHECKBOX) ||
+					Objects.equals(type, DDMFormFieldType.RADIO) ||
+					Objects.equals(type, DDMFormFieldType.SELECT) ||
+					Objects.equals(type, DDMFormFieldType.TEXT) ||
+					Objects.equals(type, DDMFormFieldType.TEXT_AREA)) {
+
+					return type;
+				}
+				else return null;
```
Comment (nhpatt) [inline modules/apps/headless-apio/structured-content/structured-content-apio-impl/src/main/java/com/liferay/structured/content/apio/internal/architect/resource/StructuredContentNestedCollectionResource.java]: ups, this if surely a bug of the source formatter

File: modules/apps/headless-apio/structured-content/structured-content-apio-test/src/testIntegration/java/com/liferay/structured/content/apio/internal/architect/resource/test/BaseStructuredContentNestedCollectionResourceTestCase.java [hunk 4]
```diff
@@ -42,6 +45,64 @@
...
+
+		Constructor constructor = innerClass.getDeclaredConstructor(
+			DDMFormFieldValue.class, DDMStructure.class);
+
+		constructor.setAccessible(true);
+
+		Object object = constructor.newInstance(
+			ddmFormFieldValue, ddmStructure);
+
+		Method method = innerClass.getMethod("getDDMFormFieldInputControl");
```
Comment (nhpatt) [inline modules/apps/headless-apio/structured-content/structured-content-apio-test/src/testIntegration/java/com/liferay/structured/content/apio/internal/architect/resource/test/BaseStructuredContentNestedCollectionResourceTestCase.java]: You can extract this, right? Or are you following the rule of 3 in duplication?

Comment (ruben-pulido) [reply modules/apps/headless-apio/structured-content/structured-content-apio-test/src/testIntegration/java/com/liferay/structured/content/apio/internal/architect/resource/test/BaseStructuredContentNestedCollectionResourceTestCase.java]: Indeed, avoiding premature refactoring

File: modules/apps/headless-apio/structured-content/structured-content-apio-impl/src/main/java/com/liferay/structured/content/apio/internal/architect/resource/StructuredContentNestedCollectionResource.java (hunk 2 above)
Comment (ruben-pulido) [reply modules/apps/headless-apio/structured-content/structured-content-apio-impl/src/main/java/com/liferay/structured/content/apio/internal/architect/resource/StructuredContentNestedCollectionResource.java]: Not needed in this case. Changed!

File: modules/apps/headless-apio/structured-content/structured-content-apio-impl/src/main/java/com/liferay/structured/content/apio/internal/architect/resource/StructuredContentNestedCollectionResource.java (hunk 3 above)
Comment (ruben-pulido) [reply modules/apps/headless-apio/structured-content/structured-content-apio-impl/src/main/java/com/liferay/structured/content/apio/internal/architect/resource/StructuredContentNestedCollectionResource.java]: Already fixed by your changes

File: modules/apps/headless-apio/structured-content/structured-content-apio-impl/src/main/java/com/liferay/structured/content/apio/internal/architect/resource/StructuredContentNestedCollectionResource.java (hunk 1 above)
Comment (ruben-pulido) [reply modules/apps/headless-apio/structured-content/structured-content-apio-impl/src/main/java/com/liferay/structured/content/apio/internal/architect/resource/StructuredContentNestedCollectionResource.java]: Good point! Changed

**Comment by ruben-pulido** [comment]
Feedback incorporated [here](https://github.com/nhpatt/liferay-portal/pull/65)

**Comment by nhpatt** [comment]
Thanks Rubén :) 🎉

**Pull request merged**